### PR TITLE
remove parse error

### PR DESCRIPTION
**Pull request by jiangxinmeng1**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue # https://github.com/matrixorigin/MO-Cloud/issues/64

## What this PR does / why we need it:
1. remove parse error
2. fix mergeblocks

### CODE REVIEW

File: pkg/vm/engine/tae/logstore/driver/batchstoredriver/file.go [hunk 1]
```diff
@@ -35,20 +35,20 @@ func MakeVersionFile(dir, name string, version uint64) string {
 	return fmt.Sprintf("%s-%d%s", filepath.Join(dir, name), version, suffix)
 }
 
-func ParseVersion(name, prefix, suffix string) (int, error) {
+func ParseVersion(name, prefix, suffix string) (n int, ok bool) {
```
Comment (fengttt) [inline pkg/vm/engine/tae/logstore/driver/batchstoredriver/file.go:38]: All file names are generated by us, why we should ever see a parse error?

File: pkg/vm/engine/tae/tables/txnentries/mergeblocks.go [hunk 2]
```diff
@@ -161,7 +163,19 @@ func (entry *mergeBlocksEntry) PrepareCommit() (err error) {
 		blks[i] = blk
 	}
 	var view *model.BlockView
+	skipped := 0
```
Comment (fengttt) [inline pkg/vm/engine/tae/tables/txnentries/mergeblocks.go:166]: This loop is VERY difficult to read.   You need to refactor it.

File: pkg/vm/engine/tae/logstore/store/checkpoint.go [hunk 3]
```diff
@@ -151,6 +151,7 @@ func (w *StoreImpl) CkpCkp() {
 	if err != nil {
 		panic(err)
 	}
+	e.Free()
```
Comment (fengttt) [inline pkg/vm/engine/tae/logstore/store/checkpoint.go:154]: I don't think here, you are doing remove parse error.    This is a big thing, you cannot slip it through a PR.
This MUST be part of PR commit message.

Comment (jiangxinmeng1) [reply pkg/vm/engine/tae/logstore/store/checkpoint.go:154]: Ok, I'll move it to another pr.

File: pkg/vm/engine/tae/logstore/driver/batchstoredriver/file.go (hunk 1 above)
Comment (jiangxinmeng1) [reply pkg/vm/engine/tae/logstore/driver/batchstoredriver/file.go:38]: There're other files in the directory.

File: pkg/vm/engine/tae/tables/txnentries/mergeblocks.go (hunk 2 above)
Comment (jiangxinmeng1) [reply pkg/vm/engine/tae/tables/txnentries/mergeblocks.go:166]: Ok, I'll refactor mergeblocks lator.

**Review by XuPeng-SH — APPROVED** [review]
lgtm

File: pkg/vm/engine/tae/logstore/driver/batchstoredriver/file.go (hunk 1 above)
Comment (fengttt) [reply pkg/vm/engine/tae/logstore/driver/batchstoredriver/file.go:38]: Which directory.   What file.   Why they land in same directory.

This does not make sense.   We control where the file is.   If the file name encode a version or something, bad file should never reach here.   This check is important -- only to catch file/dir corruption.

File: pkg/vm/engine/tae/tables/txnentries/mergeblocks.go (hunk 2 above)
Comment (fengttt) [reply pkg/vm/engine/tae/tables/txnentries/mergeblocks.go:166]: Now.   Or I won't approve.

File: pkg/vm/engine/tae/logstore/store/checkpoint.go (hunk 3 above)
Comment (fengttt) [reply pkg/vm/engine/tae/logstore/store/checkpoint.go]: Our PR commit message is annoying.   I agree.   But this is important change/fix.   Either update the PR message, or another PR is fine tool

Thank you.

**Pull request merged**